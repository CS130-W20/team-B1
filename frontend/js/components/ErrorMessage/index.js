import React from 'react';
import {Container, Row, Col} from 'react-bootstrap';

const ErrorMessage = (props) => {

  const header = props.header ? props.header : ':(';
  const message = props.message ? props.message : 'Something went wrong! Are you sure you\'re in the right place?';
  const headerSize = props.headerSize ? props.headerSize : '15em';
  const messageSize = props.messageSize ? props.messageSize : '2.5em';
  return(
    <Container>
      <Row className="justify-content-md-center">
        <Col xs lg="2" />
        <Col md="auto">
            <h1 style={{fontSize: headerSize}}>{header}</h1>
            <p style={{fontSize: messageSize}}>
              {message}
            </p>
        </Col>
        <Col xs lg="2" />
      </Row>
    </Container>
  );
};

export default ErrorMessage;
