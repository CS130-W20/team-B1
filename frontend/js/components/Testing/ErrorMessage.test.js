import React from 'react';
import renderer from 'react-test-renderer';

import ErrorMessage from '../ErrorMessage';

describe ('Testing ErrorMessage Page', () => {
    it('correctly renders', () => {
        const wrapper = renderer.create(<ErrorMessage />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});
