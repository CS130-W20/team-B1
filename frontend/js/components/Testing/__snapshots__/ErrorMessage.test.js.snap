// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ErrorMessage Page correctly renders 1`] = `
<div
  className="container"
>
  <div
    className="justify-content-md-center row"
  >
    <div
      className="col-lg-2 col"
    />
    <div
      className="col-md-auto"
    >
      <h1
        style={
          Object {
            "fontSize": "15em",
          }
        }
      >
        :(
      </h1>
      <p
        style={
          Object {
            "fontSize": "2.5em",
          }
        }
      >
        Something went wrong! Are you sure you're in the right place?
      </p>
    </div>
    <div
      className="col-lg-2 col"
    />
  </div>
</div>
`;
